### PR TITLE
Add support for auto topic creation in KIP-500

### PR DESCRIPTION
**Pull request by rondagostino**
Adds support for auto topic creation in KIP-500.  This implementation is a quick-and-dirty one that should be jettisoned in favor of https://github.com/apache/kafka/pull/9579 when that is merged to `trunk` and then to `kip-500`.  In the meantime, this PR allows KIP-500 system tests as implemented in https://github.com/confluentinc/kafka/pull/480 to pass.

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/KafkaApis.scala [hunk 1]
```diff
@@ -1189,12 +1189,23 @@ class KafkaApis(val requestChannel: RequestChannel,
...
+          throw new IllegalArgumentException(s"Illegal replication factor: $replicationFactor")
+        }
+        val topics = new CreatableTopicCollection()
+        val configs = new CreateableTopicConfigCollection()
+        properties.forEach { (k, v) => configs.add(new CreateableTopicConfig().setName(k.toString).setValue(v.toString)) }
+        topics.add(new CreatableTopic().setName(topic).setNumPartitions(numPartitions).setReplicationFactor(replicationFactor.toShort).setConfigs(configs))
+        forwardingManager.channelManager.sendRequest(
+          new CreateTopicsRequest.Builder(new CreateTopicsRequestData().setTopics(topics)),
+          (_: ClientResponse) => {})
+        metadataResponseTopic(Errors.LEADER_NOT_AVAILABLE, topic, isInternal(topic), util.Collections.emptyList())
```
Comment (cmccabe) [inline core/src/main/scala/kafka/server/KafkaApis.scala:1202]: shouldn't we wait until the topic creation succeeds, so that we match the earlier behavior?

**Comment by cmccabe** [comment]
LGTM if it works when the returned error code is something like "topic not found" to address the issue I stated above.

Long term, we'll have to reconcile this with the ideas in the KAFKA-9751 PR (and add some tests, of course)

**Comment by rondagostino** [comment]
@cmccabe It was easy enough to block, so I did that.

**Comment by cmccabe** [comment]
LGTM

**Pull request merged**